cmd/standalone/main.go

```diff
@@ -265,9 +274,12 @@ func startQuery(
 	portStr := ":" + strconv.Itoa(qOpts.Port)
 	recoveryHandler := recoveryhandler.NewRecoveryHandler(logger, true)
 	logger.Info("Starting jaeger-query HTTP server", zap.Int("port", qOpts.Port))
-	if err := http.ListenAndServe(portStr, recoveryHandler(r)); err != nil {
-		logger.Fatal("Could not launch jaeger-query service", zap.Error(err))
-	}
+	go func() {
+		defer closer.Close()
```

Comment:
I had to move this here, otherwise tests were failing. 

It's not super clear maybe we could wrap it all in the go routine